### DISCUSSION TITLE
Proposal: @mixin-to (specify @mixin from child)

### DISCUSSION THREAD

**Discussion by timkelty** [phpDocumentor/phpDocumentor#3661] [Ideas]
# Feature request

Many frameworks provide ways of dynamically loading props & methods into a class at runtime (essentially a runtime trait).

E.g., Yii Behaviors: https://www.yiiframework.com/doc/guide/2.0/en/concept-behaviors

This becomes inherently difficult for any static tooling (eg IDEs, PHPStan) to know about these additional class members.
Normally this means you need to create stubs.

`@mixin` isn't an option, as the classes we're referring to are libraries/Composer packages.

My proposal is to support something like `@mixin-to \some\ns\SomeClass` that could be added to either classes or class members to add those members to the referenced entity.

**Comment by mvriel** [comment]
What you are describing sounds a lot like `@mixin` -which doesn't do much at the moment in phpDocumentor-. See https://phpstan.org/writing-php-code/phpdocs-basics#mixins for more information on how mixins work.

`@mixin` is put on a class that consumes another and basically imports the elements from that other class into that class

**Comment by mvriel** [comment]
If the above is the case, this may be a duplicate of https://github.com/phpDocumentor/phpDocumentor/issues/1543 when I interpret it as adding support for `@mixin` to import elements. I must admit that it is desirable to add support for this, but we are focussing on keeping up with new official language additions. Thus it is not planned to do on the short term
